ios/browser/api/de_amp/de_amp_prefs.mm

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/api/de_amp/de_amp_prefs+private.h"
+
+#include "brave/components/de_amp/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "ios/chrome/browser/shared/model/application_context/application_context.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+
+@implementation DeAmpPrefs {
```

Comment:
shouldn't we just expose a generic interface to prefs? cc @Brandon-T

Reply:
I followed along other examples we had but I agree some generic approach would be perhaps nicer.

Reply:
can we look at this in a follow-up?

Reply:
Personally I think it's a good idea. But while I opened up some discussion around a generic approach there was no conclusive decision about this.  Im willing to re-open this discussion more openly.